lib/greenhouse_api/client.rb

```diff
@@ -33,18 +33,40 @@ def get_current_offer_for_application(application_id)
         http_method: :get,
         headers: headers,
         endpoint: "applications/#{application_id}/offers/current_offer",
-        params: {}
       )
+      create_response(response)
+    end
 
-      if response.status == 200
-        Response.new(
-          body: response.body,
-          headers: response.headers,
-          status: response.status
-        )
-      else
-        response
-      end
+    def create_user(on_behalf_of, body)
+      response = request(
+          http_method: :post,
+          headers: headers(on_behalf_of),
+          endpoint: "users",
+          body: body
+      )
+      create_response(response, 201)
+    end
+
+    def disable_user(on_behalf_of, body)
+      response = request(
+          http_method: :patch,
+          headers: headers(on_behalf_of),
+          endpoint: "users/disable",
+          body: body,
+          api_version: 'v2'
+      )
+      create_response(response)
+    end
+
+    def enable_user(on_behalf_of, body)
```

Comment:
This client class is getting big, do you think we should reorganize so that it's more maintainable and easier for the users. A few ideas I have are

1. move code into modules and mix them in this class. This results in the same api as we currently have
2. move into separate classes each deals with a resource (candidates, offers, users, etc.) Change the api such that it's clear which resources we're interacting with, e.g. `client.users.enable_user`

We can also do it in a separate PR, but I think we should only bump the version until we solidify the interface

Reply:
yeah great thinking
I was looking at some other api client gems and another option is using inheritance. so we could follow what the ruby google api sdk does like
```ruby
```

Reply:
Agreed, I'll add a story for this 👍